pkg/controller/backupbucket/backupbucket.go

```diff
@@ -0,0 +1,135 @@
+	"github.com/onmetal/gardener-extension-provider-onmetal/pkg/onmetal"
+	storagev1alpha1 "github.com/onmetal/onmetal-api/api/storage/v1alpha1"
+)
+
+// ensureBackupBucket creates onmetal backupBucket object and returns access to bucket
+func (a *actuator) ensureBackupBucket(ctx context.Context, namespace string, onmetalClient client.Client, backupBucket *extensionsv1alpha1.BackupBucket) error {
+	bucket := &storagev1alpha1.Bucket{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      backupBucket.Name,
+			Namespace: namespace,
+		},
+		Spec: storagev1alpha1.BucketSpec{
+			BucketClassRef: &corev1.LocalObjectReference{
+				Name: a.backupBucketConfig.BucketClassName,
+			},
+		},
+	}
+	//create onmetal bucket
+	if _, err := controllerutil.CreateOrPatch(ctx, onmetalClient, bucket, nil); err != nil {
+		return fmt.Errorf("failed to create or patch backup bucket %s: %w", client.ObjectKeyFromObject(bucket), err)
+	}
+	//wait for bucket creation
+	if err := wait.PollUntilWithContext(ctx, 1*time.Second, func(ctx context.Context) (done bool, err error) {
+		err = onmetalClient.Get(ctx, client.ObjectKey{Namespace: bucket.Namespace, Name: bucket.Name}, bucket)
+		if err == nil && bucket.Status.State == storagev1alpha1.BucketStateAvailable && isBucketAccessDetailsAvailable(bucket) {
+			return true, nil
+		}
+		return false, nil
+	}); err != nil {
+		return fmt.Errorf("could not determine status of backup bucket")
+	}
+
+	accessSecret := &corev1.Secret{}
+	if err := onmetalClient.Get(ctx, client.ObjectKey{Namespace: namespace, Name: bucket.Status.Access.SecretRef.Name}, accessSecret); err != nil {
+		return fmt.Errorf("error bucket access secret")
+	}
+	//update backupBucket secret
+	if err := a.updateBackupBucketStatus(ctx, backupBucket, accessSecret.Data, bucket.Status.Access.Endpoint); err != nil {
+		return fmt.Errorf("error while updating backupbucket status")
+	}
+	return nil
+}
+
+func isBucketAccessDetailsAvailable(bucket *storagev1alpha1.Bucket) bool {
+	return bucket.Status.Access != nil && bucket.Status.Access.SecretRef != nil && bucket.Status.Access.Endpoint != ""
+}
+
+// updateBackupBucketStatus updates backupBucket status with access secretRef
+func (a *actuator) updateBackupBucketStatus(ctx context.Context, backupBucket *extensionsv1alpha1.BackupBucket, secretData map[string][]byte, endpoint string) error {
+	if secretData == nil {
+		return fmt.Errorf("secret does not contain any data")
+	}
+
+	accessKeyID, ok := secretData[onmetal.BucketAccessKeyID]
+	if !ok {
+		return fmt.Errorf("missing %q field in secret", onmetal.BucketAccessKeyID)
+	}
+
+	secretAccessKey, ok := secretData[onmetal.BucketSecretAccessKey]
+	if !ok {
+		return fmt.Errorf("missing %q field in secret", onmetal.BucketSecretAccessKey)
+	}
+
+	accessSecretData := map[string][]byte{}
+	accessSecretData[onmetal.AccessKeyID] = []byte(accessKeyID)
+	accessSecretData[onmetal.SecretAccessKey] = []byte(secretAccessKey)
+	accessSecretData[onmetal.Endpoint] = []byte(endpoint)
```

Comment:
We might want to use here constants which are defined by the [`gardener/etcd-backup-restore`](https://github.com/gardener/etcd-backup-restore) project. That way we ensure, that we store the credentials in the correct keys.

Reply:
@afritzler I am not finding any constants in [gardener/etcd-backup-restore](https://github.com/gardener/etcd-backup-restore) repo. Can you point out from where I can refer?

Reply:
I just had a look and there don't seem to be any. :(